File: 11723.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

int	checker(char *signal, int *set)
{
	int	i = 0, num;
	char *add = "add";
	char *remove = "remove";
	char *check = "check";
	char *toggle = "toggle";
	char *all = "all";
	char *empty = "empty";

	scanf("%d", &num);
	if (strcmp(add, signal) == 0)// 명령 부분이 똑같으면
	{
		set[num]++;
		return (-1);
	}
	else if (strcmp(remove, signal) == 0)
	{
		set[num] = 0;
		return (-1);
	}
	else if (strcmp(check, signal) == 0)
	{
		if (set[num] == 0)
			return (0);
		else
			return (1);
	}
	else if (strcmp(toggle, signal) == 0)
	{
		if (set[num] == 0)
			set[num] = 1;
		else
			set[num] = 0;
		return (-1);
	}
	else if (strcmp(all, signal) == 0)
	{
		i = 0;
		while (++i < 21)
			set[i] = 1;
		return (-1);
	}
	else if (strcmp(empty, signal) == 0)
	{
		i = 0;
		while (++i < 21)
			set[i] = 0;
		return (-1);
	}
}

int	main()
{
	int	n, i = -1, check;
	int	set[21] = { 0 };
	char signal[10];
	scanf("%d", &n);

	while (++i < n)
	{
		scanf("%s", signal);
		check = checker(signal, set);
		if (check == 1)
			printf("1\n");
		else if (check == 0)
			printf("0\n");
	}
}

File: 11723.py
import sys
input = sys.stdin.readline

n = int(input())
res = set()
for i in range(n):
    cmd = input().split()
    if cmd[0] == "add":
        res.add(int(cmd[1]))
 
    elif cmd[0] == "remove":
        if int(cmd[1]) in res:
            res.remove(int(cmd[1]))

    elif cmd[0] == "check":
        if int(cmd[1]) in res:
            print(1)
        else :
            print(0)

    elif cmd[0] == "toggle":
        if int(cmd[1]) in res:
            res.remove(int(cmd[1]))
        else:
            res.add(int(cmd[1]))    

    elif cmd[0] == "all":
        res = set([i for i in range(1, 21)])
  
    elif cmd[0] == "empty":
        res.clear() #clear 리스트 내의 모든 원소 삭제


    
        